### ISSUE TITLE
MLPs apply `torch.nn.CrossEntropyLoss` on `log_softmax` outputs

### ISSUE THREAD

**Issue by liqilei** [enriquetomasmb/nebula#15]
Thanks for the code. Just a hand-up that currently the MLP networks apply  `x = torch.log_softmax(x, dim=1)` then proceed to `torch.nn.CrossEntropyLoss()` which internally combines `log_softmax` and `NLLLoss`. Therefore, it would be better to use NLL loss or remove the log_softmax operation.

**Comment by enriquetomasmb** [comment]
Thanks, @liqilei, you are absolutely correct. I just updated the MLP models, the commit cb478b0 includes the changes. 

Next time, I encourage you to contribute any improvement or changes through PR. It is always nice to collaborate with other researchers and grow the platform from different perspectives.

Best.
